src/safeposix/syscalls/net_calls.rs

```diff
@@ -2185,19 +2266,27 @@ impl Cage {
         }
     }
 
+    // this function is an inner function of shutdown and checks for fd
     pub fn _cleanup_socket(&self, fd: i32, how: i32) -> i32 {
+        // get the file descriptor object
         let checkedfd = self.get_filedescriptor(fd).unwrap();
         let mut unlocked_fd = checkedfd.write();
         if let Some(ref mut filedesc_enum) = &mut *unlocked_fd {
             let inner_result = self._cleanup_socket_inner(filedesc_enum, how, true);
```

Comment:
is the third argument ever set to false instead of true?

Reply:
for _cleanup_socket_inner, this is the only place where we call it, so yeah, this is always called with true for now. But for _cleanup_socket_inner_helper, it may be called with shutdown set to false when sockethandle is dropped